src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/ProjectSystems/VsMSBuildProjectSystem.cs

```
@@ -990,17 +711,267 @@ public IEnumerable<string> GetFullPaths(string fileName)
+
+            try
+            {
+                // Get the reference name without extension
+                var referenceName = Path.GetFileNameWithoutExtension(name);
+
+                // Remove the reference from the project
+                // NOTE:- Project.Object.References.Item requires Reference.Identity
+                //        which is, the Assembly name without path or extension
+                //        But, we pass in the assembly file name. And, this works for
+                //        almost all the assemblies since Assembly Name is the same as the assembly file name
+                //        In case of F#, the input parameter is case-sensitive as well
+                //        Hence, an override to THIS function is added to take care of that
+                var reference = References.Item(referenceName);
+                if (reference != null)
+                {
+                    reference.Remove();
+                    NuGetProjectContext.Log(ProjectManagement.MessageLevel.Debug, Strings.Debug_RemoveReference, name, ProjectName);
+                }
+            }
+            catch (Exception e)
+            {
+                NuGetProjectContext.Log(ProjectManagement.MessageLevel.Warning, e.Message);
+            }
+        }
+
+        public virtual async Task<bool> ReferenceExistsAsync(string name)
+        {
+            await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+            try
+            {
+                var referenceName = name;
+                if (ProjectManagement.Constants.AssemblyReferencesExtensions.Contains(Path.GetExtension(name), StringComparer.OrdinalIgnoreCase))
+                {
+                    // Get the reference name without extension
+                    referenceName = Path.GetFileNameWithoutExtension(name);
+                }
+
+                return References.Item(referenceName) != null;
+            }
+            catch
```

Why not log this exception?

---

I think there were a good reason no to log (too noisy?). It's a reference query that should be swallowing the exception.